pkg/operator/ceph/object/notification/obc_label_controller.go

```diff
@@ -149,36 +152,70 @@ func (r *ReconcileOBCLabels) reconcile(request reconcile.Request) (reconcile.Res
 				logger.Warningf("bucket notification label mismatch. ignoring key %q value %q", labelKey, labelValue)
 				continue
 			}
-			// for each notification label fetch the bucket notification CRD
-			logger.Debugf("bucket notification label %q found on ObjectbucketClaim %q", labelValue, bucketName)
-			notification := &cephv1.CephBucketNotification{}
-			bnName := types.NamespacedName{Namespace: obc.Namespace, Name: labelValue}
-			if err := r.client.Get(r.opManagerContext, bnName, notification); err != nil {
-				if kerrors.IsNotFound(err) {
-					logger.Infof("CephBucketNotification %q not found", bnName)
-					return waitForRequeueIfNotificationNotReady, nil
+			labelList = append(labelList, labelValue)
+			found := false
+			for _, existingValue := range bnList {
+				if labelValue == existingValue {
+					found = true
+					break
 				}
-				return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve CephBucketNotification %q", bnName)
 			}
-
-			// get the topic associated with the notification, and make sure it is provisioned
-			topicName := types.NamespacedName{Namespace: notification.Namespace, Name: notification.Spec.Topic}
-			bucketTopic, err := topic.GetProvisioned(r.client, r.opManagerContext, topicName)
-			if err != nil {
-				logger.Infof("CephBucketTopic %q not provisioned yet", topicName)
-				return waitForRequeueIfTopicNotReady, nil
+			// add it into new list for creating notifications for the bucket
+			if !found {
+				addList = append(addList, labelValue)
 			}
-
-			if err = validateObjectStoreName(bucketTopic, objectStoreName); err != nil {
-				return reconcile.Result{}, err
+		}
+	}
+	// remove notifications which are no longer specified in the OBC labels
+	for _, oldValue := range bnList {
```

Comment:
I think this only works if `bnList` is the list of notifications as known by the RGW. Is that what `bnList` represents?

Reply:
Yes. bnList represents bucket notifications on listed on the backend